api.go

```diff
@@ -215,25 +215,40 @@ func (h *Headscale) PollNetMapHandler(c *gin.Context) {
 		return
 	}
 
-	log.Printf("[%s] sending initial map", m.Name)
-	pollData <- *data
-
 	// We update our peers if the client is not sending ReadOnly in the MapRequest
 	// so we don't distribute its initial request (it comes with
 	// empty endpoints to peers)
-	if !req.ReadOnly {
-		peers, _ := h.getPeers(m)
-		h.pollMu.Lock()
-		for _, p := range *peers {
-			log.Printf("[%s] notifying peer %s (%s)", m.Name, p.Name, p.Addresses[0])
-			if pUp, ok := h.clientsPolling[uint64(p.ID)]; ok {
-				pUp <- []byte{}
-			} else {
-				log.Printf("[%s] Peer %s does not appear to be polling", m.Name, p.Name)
-			}
+
+	// Details on the protocol can be found in https://github.com/tailscale/tailscale/blob/main/tailcfg/tailcfg.go#L696
+	log.Printf("[%s] ReadOnly=%t   OmitPeers=%t    Stream=%t", m.Name, req.ReadOnly, req.OmitPeers, req.Stream)
+
+	if req.ReadOnly {
+		log.Printf("[%s] Client is starting up. Asking for DERP map", m.Name)
+		c.Data(200, "application/json; charset=utf-8", *data)
+		return
+	}
+	if req.OmitPeers {
+		log.Printf("[%s] Client is starting up. Ready to receive the peers", m.Name)
```

Comment:
Based on the comments from the tailscale source, this log line isn't quite right. How about something like this instead:

log.Printf("[%s] Client sent endpoint update and is ok with a response without peer list.", m.Name)

Arguably this log messing is kind of meaningless in day to day operation; this would be a good candidate for a debug level log message I think.

Reply:
Fully agree. This messages should be disabled once we are confident with our interpretation of the protocol.

Reply:
Fully agree. This messages should be disabled once we are confident with our interpretation of the protocol.